src/hotspot/os/posix/os_posix.cpp

```diff
@@ -349,13 +329,39 @@ char* os::reserve_memory_aligned(size_t size, size_t alignment, int file_desc) {
       os::release_memory(extra_base + begin_offset + size, end_offset);
```

Comment:
Probably yes. But if you don't mind, I would leave that as is. A nice picture above provides some clarification and I'd really like to avoid touching unrelated code

Reply:
Sure.